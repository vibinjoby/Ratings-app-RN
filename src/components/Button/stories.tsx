import React from 'react'
import { View, StyleSheet } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import Button from './component'
import Colors from '../../utilities/colors'

const ButtonComponent = () => (
  <View style={styles.container}>
    <Button
      title={'GET STARTED'}
      onPress={() => console.log('pressed')}
      customStyle={styles.getStartedBtn}
    />
    <View style={styles.buttonDock}>
      <Button
        title={'Login'}
        textColor={Colors.black}
        buttonColor={Colors.lightGrey}
        customStyle={styles.leftBtn}
        onPress={() => console.log('pressed')}
      />
      <Button
        title={'Sign Up'}
        textColor={Colors.white}
        customStyle={styles.rightBtn}
        buttonColor={Colors.appOrange}
        onPress={() => console.log('pressed')}
      />
    </View>
  </View>
)

storiesOf('Components', module).add('Button', () => <ButtonComponent />)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    justifyContent: 'space-around',
  },
  buttonDock: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  getStartedBtn: { borderRadius: 19 },
  leftBtn: {
    flex: 1,
    marginRight: 10,
  },
  rightBtn: {
    flex: 1,
  },
})
