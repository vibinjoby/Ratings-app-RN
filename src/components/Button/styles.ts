import { StyleSheet } from 'react-native'
import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.appGreen,
    padding: 14,
    alignItems: 'center',
    borderRadius: 30,
  },
  btnTxt: {
    ...Typography.Body.regular,
    color: Colors.white,
  },
})
