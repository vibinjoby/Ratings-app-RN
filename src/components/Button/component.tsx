import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import Colors from '../../utilities/colors'

import styles from './styles'

export interface ButtonProps {
  testID?: string
  title: string
  buttonColor?: string
  textColor?: string
  customStyle?: Record<string, unknown>
  disabled?: boolean
  onPress: () => void
}

const Button: React.FC<ButtonProps> = ({
  testID,
  title,
  customStyle,
  buttonColor,
  textColor,
  disabled,
  onPress,
}: ButtonProps) => {
  return (
    <TouchableOpacity
      disabled={disabled}
      onPress={onPress}
      activeOpacity={0.8}
      testID={testID}
      style={[
        styles.container,
        buttonColor ? { backgroundColor: buttonColor } : {},
        customStyle,
        disabled ? { backgroundColor: Colors.borderColor } : {},
      ]}
    >
      <Text style={[styles.btnTxt, textColor ? { color: textColor } : {}]}>{title}</Text>
    </TouchableOpacity>
  )
}

export default Button
