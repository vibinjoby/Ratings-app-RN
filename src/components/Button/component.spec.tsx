import 'react-native'
import React from 'react'
import Button from './component'

import { render, fireEvent } from '@testing-library/react-native'

describe('Button', () => {
  const callbackFn = jest.fn()
  const mockTitle = 'title'

  it('should match snapshot', () => {
    const { toJSON } = render(<Button title={mockTitle} onPress={callbackFn} />)
    expect(toJSON).toMatchSnapshot()
  })

  it('should be able to click the button', () => {
    const { getByTestId } = render(
      <Button title={mockTitle} onPress={callbackFn} testID="buttonContainer" />,
    )
    fireEvent.press(getByTestId('buttonContainer'))
    expect(callbackFn).toBeCalled()
  })

  it('should be render button text', () => {
    const { getByText } = render(<Button title={mockTitle} onPress={callbackFn} />)
    expect(getByText(mockTitle)).toBeTruthy()
  })
})
