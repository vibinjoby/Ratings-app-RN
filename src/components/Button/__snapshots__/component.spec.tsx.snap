// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `[Function]`;
